packages/core/src/index.ts

```diff
@@ -274,6 +274,11 @@ class Hopr extends EventEmitter {
     await this.connector.start()
     verbose('Started HoprEthereum. Waiting for indexer to find connected nodes.')
 
+    // Add us as public node if announced
+    if (this.options.announce) {
```

Comment:
Question on an edge case: What if this node was announced before, but restart without an `--announce` flag? Should such a node be "GREEN"?

Reply:
Good point. This case should handled by the `onPeerAnnouncement` which also populates the `knowPublicNodesCache`.
So basically, if I announced myself in the past and restart without the `--announce` flag, the `onPeerAnnouncement` should be called for my peer ID.

Reply:
Per discussion, the case when user previously announced their node, but later stops using the `--announce` flag in an attempt to conceal the node will be fully resolved by #4323 .